### PR TITLE
Feature/sre 3108 handle vs deleted

### PR DESCRIPTION
**Pull request by activehigh**
### Background ###

- Added a build pipeline
- Added vs watcher to make removal and update logic more stable
- Added not found check for vs

### JIRA Ticket ###

-  [SRE-3108](https://rokt.atlassian.net/browse/SRE-3108)

### Checklist: ###

- [x] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [x] I have verified that CI completes successfully.
- [ ] All insignificant commits have been squashed.

### CODE REVIEW

File: controller/reconciler.go [hunk 1]
```diff
@@ -42,11 +42,11 @@ func Reconcile(ctx reconciler.Context, client *versionedclient.Clientset, patch
 		}
 	} else if oputil.Contains(patch.Finalizers, finalizerName) {
 		if err := updateTarget(client, patch, true); err != nil {
-			if kerr.IsNotFound(err) {
-				// do not need to panic just log output
-				ctx.Logger().Info("Virtual service not found.")
+			if !kerr.IsNotFound(err) {
+				// ignore if virtuals service is not found
```
Comment (AaronChelvan) [inline controller/reconciler.go:46]: I think this comment should be in the else statement

File: controller/reconciler.go [hunk 2]
```diff
@@ -57,7 +57,12 @@ func Reconcile(ctx reconciler.Context, client *versionedclient.Clientset, patch
 	}
 	if patch.ResourceVersion != patch.Status.HandledRevision {
 		if err := updateTarget(client, patch, false); err != nil {
-			return err
+			if !kerr.IsNotFound(err) {
+				// ignore if virtuals service is not found
```
Comment (AaronChelvan) [inline controller/reconciler.go:61]: I think this comment should be in the else statement

**Comment by AaronChelvan** [comment]
You've also committed the `istio-virtualservice-merger` binary

File: controller/reconciler.go [hunk 3]
```diff
@@ -42,11 +42,11 @@ func Reconcile(ctx reconciler.Context, client *versionedclient.Clientset, patch
 		}
 	} else if oputil.Contains(patch.Finalizers, finalizerName) {
 		if err := updateTarget(client, patch, true); err != nil {
-			if kerr.IsNotFound(err) {
-				// do not need to panic just log output
-				ctx.Logger().Info("Virtual service not found.")
+			if !kerr.IsNotFound(err) {
```
Comment (alexnguyenatwork) [inline controller/reconciler.go]: seems like you can flip the statement bodies and not need the `!` here? double negatives are harder to follow
i.e. `if kerr.IsNotFound(err) { ...`

**Review by alexnguyenatwork — APPROVED** [review]

**Review by alexnguyenatwork — APPROVED** [review]

**Review by SooPark — APPROVED** [review]

**Pull request merged**